### PR TITLE
Complete ChannelPromise for Http2WindowUpdateFrames in Http2FrameCodec.

### PR DESCRIPTION
**Pull request by buchgr**
Motivation:

The channel promise of a window update frame is not completed correctly,
depending on the failure or success of the operation.

Modification:

Succeed / Fail the promise if the window update succeeds / fails.

Result:

Correctly succeed / fail the promise.

### CODE REVIEW

File: codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java [hunk 1]
```diff
@@ -155,13 +154,18 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
...
 
-    private void consumeBytes(int streamId, int bytes) {
+    private void consumeBytes(int streamId, int bytes, ChannelPromise promise) {
         try {
             Http2Stream stream = http2Handler.connection().stream(streamId);
             http2Handler.connection().local().flowController()
                         .consumeBytes(stream, bytes);
         } catch (Throwable t) {
+            promise.setFailure(t);
             exceptionCaught(ctx, t);
```
Comment (normanmaurer) [inline codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java]: @buchgr should we remove the `exceptionCaught(...)` here ? I think if we propagate to the promise we should not also call `exceptionCaught(...)`. Also @nmittler @Scottmitch WDYT ?

**Comment by buchgr** [comment]
cc @olix0r

**Comment by normanmaurer** [comment]
@Scottmitch PTAL as well.

**Comment by Scottmitch** [comment]
thanks @buchgr ! 1 comment and lgtm

**Comment by normanmaurer** [comment]
+1

@buchgr please address ... After that I will pull in

**Comment by buchgr** [comment]
PTAL.

**Comment by buchgr** [comment]
Test failures seem unrelated:  
```
EpollSocketConnectionAttemptTest.testConnectCancellation 
```

**Comment by normanmaurer** [comment]
@buchgr yep some dns problems... Let me trigger again

**Comment by normanmaurer** [comment]
@buchgr thanks a lot buddy!

Cherry-picked as 328510468c3da0f6a37d9710d7723f817eff8ce8

**Pull request closed without merging**